src/uipath_mcp/_cli/_runtime/_factory.py

```diff
@@ -0,0 +1,145 @@
+"""Factory for creating MCP runtime instances."""
+
+import json
+import logging
+import os
+import uuid
+from typing import Any
+
+from uipath.runtime import (
+    UiPathRuntimeContext,
+    UiPathRuntimeProtocol,
+)
+from uipath.runtime.errors import UiPathErrorCategory
+
+from uipath_mcp._cli._runtime._exception import McpErrorCode, UiPathMcpRuntimeError
+from uipath_mcp._cli._runtime._runtime import UiPathMcpRuntime
+from uipath_mcp._cli._utils._config import McpConfig
+
+logger = logging.getLogger(__name__)
+
+
+class UiPathMcpRuntimeFactory:
+    """Factory for creating MCP runtimes from mcp.json configuration."""
+
+    def __init__(
+        self,
+        context: UiPathRuntimeContext,
+    ):
+        """Initialize the factory.
+
+        Args:
+            context: UiPathRuntimeContext to use for runtime creation.
+        """
+        self.context = context
+        self._mcp_config: McpConfig | None = None
+        self._server_id: str | None = None
+        self._server_slug: str | None = None
+
+        # Load fps context from uipath.json if available
+        self._load_fps_context()
+
+    def _load_fps_context(self) -> None:
+        """
+        Load fps context from uipath.json for server registration.
+        """
+        config_path = self.context.config_path or "uipath.json"
```

Comment:
```suggestion
        config_path = self.context.config_path or UiPathConfig.config_file_path()
```
ref: https://github.com/UiPath/uipath-python/blob/b2eed6e5d8322855144db0d3b71d2c219d1bdb71/src/uipath/platform/common/_config.py#L27